diff --git a/src/managers/SeatManager.cpp b/src/managers/SeatManager.cpp
index fc14f0fc..730b6e4c 100644
--- a/src/managers/SeatManager.cpp
+++ b/src/managers/SeatManager.cpp
@@ -313,7 +313,7 @@ void CSeatManager::sendPointerFrame(WP<CWLSeatResource> pResource) {
     }
 }
 
-void CSeatManager::sendPointerAxis(uint32_t timeMs, wl_pointer_axis axis, double value, int32_t discrete, int32_t value120, wl_pointer_axis_source source,
+void CSeatManager::sendPointerAxis(uint32_t timeMs, wl_pointer_axis axis, double value, int32_t discrete, wl_pointer_axis_source source,
                                    wl_pointer_axis_relative_direction relative) {
     if (!state.pointerFocusResource)
         return;
@@ -331,7 +331,6 @@ void CSeatManager::sendPointerAxis(uint32_t timeMs, wl_pointer_axis axis, double
             p->sendAxisRelativeDirection(axis, relative);
 
             if (source == 0) {
-                p->sendAxisValue120(axis, value120);
                 p->sendAxisDiscrete(axis, discrete);
             }
 
diff --git a/src/managers/SeatManager.hpp b/src/managers/SeatManager.hpp
index 43ebe8b5..f22160b4 100644
--- a/src/managers/SeatManager.hpp
+++ b/src/managers/SeatManager.hpp
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <cstdint>
 #include <wayland-server-protocol.h>
 #include "../helpers/WLListener.hpp"
 #include "../macros.hpp"
@@ -66,8 +65,7 @@ class CSeatManager {
     void     sendPointerButton(uint32_t timeMs, uint32_t key, wl_pointer_button_state state);
     void     sendPointerFrame();
     void     sendPointerFrame(WP<CWLSeatResource> pResource);
-    void     sendPointerAxis(uint32_t timeMs, wl_pointer_axis axis, double value, int32_t discrete, int32_t value120, wl_pointer_axis_source source,
-                             wl_pointer_axis_relative_direction relative);
+    void     sendPointerAxis(uint32_t timeMs, wl_pointer_axis axis, double value, int32_t discrete, wl_pointer_axis_source source, wl_pointer_axis_relative_direction relative);
 
     void     sendTouchDown(SP<CWLSurfaceResource> surf, uint32_t timeMs, int32_t id, const Vector2D& local);
     void     sendTouchUp(uint32_t timeMs, int32_t id);
diff --git a/src/managers/input/InputManager.cpp b/src/managers/input/InputManager.cpp
index 81a46f97..644ace81 100644
--- a/src/managers/input/InputManager.cpp
+++ b/src/managers/input/InputManager.cpp
@@ -1,7 +1,6 @@
 #include "InputManager.hpp"
 #include "../../Compositor.hpp"
 #include "wlr/types/wlr_switch.h"
-#include <cstdint>
 #include <ranges>
 #include "../../config/ConfigValue.hpp"
 #include "../../desktop/Window.hpp"
@@ -798,9 +797,8 @@ void CInputManager::onMouseWheel(IPointer::SAxisEvent e) {
             }
         }
     }
-    double deltaDiscrete = factor * e.deltaDiscrete / std::abs(e.deltaDiscrete);
-    g_pSeatManager->sendPointerAxis(e.timeMs, e.axis, factor * e.delta, deltaDiscrete > 0 ? std::ceil(deltaDiscrete) : std::floor(deltaDiscrete),
-                                    std::round(factor * e.deltaDiscrete), e.source, WL_POINTER_AXIS_RELATIVE_DIRECTION_IDENTICAL);
+
+    g_pSeatManager->sendPointerAxis(e.timeMs, e.axis, factor * e.delta, std::round(factor * e.deltaDiscrete / 120), e.source, WL_POINTER_AXIS_RELATIVE_DIRECTION_IDENTICAL);
 }
 
 Vector2D CInputManager::getMouseCoordsInternal() {
